src/form/Button.js

```diff
@@ -0,0 +1,87 @@
+"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/HTMLWidget", "../api/IInput", "css!./Input"], factory);
+    } else {
+        root.form_Button = factory(root.d3, root.common_HTMLWidget, root.api_IInput);
+    }
+}(this, function (d3, HTMLWidget, IInput) {
+    function Button() {
+        HTMLWidget.call(this);
+        IInput.call(this);
+
+        this._tag = "div";
+        this._inputElement = [];
+    }
+    Button.prototype = Object.create(HTMLWidget.prototype);
+    Button.prototype.constructor = Button;
+    Button.prototype._class += " form_Button";
+    Button.prototype.implements(IInput.prototype);
+
+    Button.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        this.createInput(element);
+    };
+
+    Button.prototype.disable = function (disable) {
+        this._inputElement.forEach(function(e, idx) {
+            e.attr("disabled", disable ? "disabled" : null);
+        });
+    };
+
+    Button.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+
+        var context = this;
+        this._inputElement.forEach(function(e) {
+            e.attr("name", context.name());
+        });
+
+        this.checkNodeName("BUTTON", element);
+        this._inputElement[0].text(this.value());
+    };
+
+    Button.prototype.createInput = function (element) {
```

Comment:
Yes - why do we have a "createInput" method?  The code it contains is typically the code we have in the "enter" method, so would be better there?